src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,612 @@
+
+                        # TODO add metadata from queue into mergaMetadata function so that metadata is correct.
+                        if DATA_CONTENT_TO_ASM[dataContent] == None:
+                            da = model.DataArray(numpy.array([[0]], dtype=numpy.uint8), metadata=self._mergeMetadata())
+                        else:
+                            # TODO add functionality for getting a full and thumbnail image if simulator is updated
+                            logging.error("Option %s not yet implemented" % meta_info[1])
+                            img = self.parent.ASMAPICall(self._server_URL + "/scan/field", _METHOD_GET, 200,
+                                                         (field_data.position_x,
+                                                          field_data.position_y,
+                                                          DATA_CONTENT_TO_ASM[dataContent].to_dict()))
+                            da = model.DataArray(img, metadata=self._mergeMetadata())
+
+                        notifier_func(da)
+
+                    else:
+                        logging.warning("Start ASM acquisition before taking field images")
+
+                elif command == "stop":
+                    if self._acquisition_in_progress:
+                        self._acquisition_in_progress = False
+                        self.parent.ASMAPICall(self._server_URL + "/scan/finish_mega_field", _METHOD_POST, 204)
+                    else:
+                        logging.warning("ASM acquisition was already at status '%s'" % command)
+
+                elif command == "terminate":
+                    self._acquisition_in_progress = None
+                    # self.parent.ASMAPICall(self._server_URL + "/scan/finish_mega_field", _METHOD_POST, 204)
+                    # time.sleep(0.5)
+                    raise Exception("Terminated acquisition")
+
+                else:
+                    logging.error("Received invalid command '%s' is skipped" % command)
+                    raise ValueError
+
+                tnow = time.time()
+                # sleep a bit to avoid refreshing too fast
+                tsleep = tnext - tnow
+                if tsleep > 0.01:
+                    time.sleep(tsleep)
+
+                tnext = time.time() + 0.2  # max 5 Hz
+
+        except Exception as e:
+            if command == "terminate":
+                raise Exception("Terminated acquisition")
+
+            elif command is not None:
+                logging.error("last message was not executed, should have perfomed action: '%s'\n"
+                              "received exepction: \n %s" % (command, e))
+        finally:
+            self.parent.ASMAPICall(self._server_URL + "/scan/finish_mega_field", _METHOD_POST, 204)
+            logging.debug("Acquisition thread ended")
+
+    def start_acquisition(self):
+        megafield_metadata = self._assemble_megafield_metadata()
```

Comment:
Typically, you want to check whether the acq_thread is running, and if not, start it. This way, if something bad happened, just stopping/starting the acquisition will fix it (instead of having to restart the whole backend). See the avantes driver (`_generator`) for a clean way do write it.